src/parse_cmd.rs

```diff
@@ -0,0 +1,98 @@
+pub trait ToCmd<'a> {
+    fn to_cmd(&'a self) -> Vec<String>;
+}
+
+impl<'a> ToCmd<'a> for str {
+    fn to_cmd(&'a self) -> Vec<String> {
+        let mut args = Vec::new();
+        let mut current_arg = String::new();
+        let mut in_quote = Vec::new();
+
+        for c in self.chars() {
+            if in_quote.is_empty() && c.is_whitespace() {
+                args.push(current_arg);
+                current_arg = String::new();
+                continue;
+            }
+
+            current_arg.push(c);
+
+            if c == '"' || c == '\'' {
+                if in_quote.last() == Some(&c) {
+                    in_quote.pop();
+                } else {
+                    in_quote.push(c);
+                }
+            }
```

Comment:
Ich glaube es gibt Randfälle, in denen sich das noch unerwartet verhalten könnte. Ich spiele mal Advocatus Diaboli:

```rust
assert_eq!(
    r#"echo lorem                           ipsum"#.to_cmd(),
    vec!["echo", "lorem", "ipsum"]
);
assert_eq!(
    r#"echo "lorem' ipsum" dolor "sit' amet" '"#.to_cmd(),
    vec!["echo", "lorem' ipsum", "dolor", "sit' amet"]
);
assert_eq!(
    r#"echo "lorem\" ipsum" dolor "sit\" amet"'"#.to_cmd(),
    vec!["echo", "lorem\" ipsum", "dolor", "sit\" amet"]
);
```

Bei echo ist das unkritisch, bei anderen Befehlen könnte es schon problematisch sein, wenn Argument 3 plötzlich an Position 4 steht, etc. Vielleicht kann man sich hier auch irgendein fuzzy-testing überlegen? Das macht es natürlich alles sehr sehr kompliziert. Wichtig wäre aber, dass es sich vorhersehbar verhält.

M.E. wäre folgende Invariante sehr erstrebenswert:
* Wenn `"..."` ein gültiger Rust String ist (`...` sind Platzhalter), dann sollte `stringify!("...").to_cmd()]` gleich `["..."].to_cmd()` sein.

Ich kenne die `stringify!` Methode nicht im Detail, vermute aber, dass auf einen String angewendet, `JSON::from_str(stringify!("..."))` weitestgehend die Identität sein sollte. Auf dieser Annahme basiert ja die aktuelle Makro-Logik. Auch hier wäre vielleicht mal eine genauere Recherche oder Fuzzy Testing sinnvoll. :-)